### PR TITLE
Update liquid clustering docs

### PR DESCRIPTION
**Pull request by zedtang**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (docs)

## Description

Add docs for OPTIMIZE FULL, in-place migration, and create table from external location.
## How was this patch tested?

127 0 0 1_8000_delta-clustering html (6)

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: docs/source/delta-clustering.md [hunk 1]
```diff
@@ -34,7 +34,8 @@ To enable liquid clustering, add the `CLUSTER BY` phrase to a table creation sta
   CREATE TABLE table1(col0 int, col1 string) USING DELTA CLUSTER BY (col0);
 
   -- Using a CTAS statement
-  CREATE TABLE table2 CLUSTER BY (col0)  -- specify clustering after table name, not in subquery
+  CREATE EXTERNAL TABLE table2 CLUSTER BY (col0)  -- specify clustering after table name, not in subquery
+  LOCATION 'table_location'
```
Comment (rahulsmahadev) [inline docs/source/delta-clustering.md:38]: do we need location here, previously it was not there

Comment (zedtang) [reply docs/source/delta-clustering.md:38]: yes, we now supports creating table from external location in https://github.com/delta-io/delta/pull/3251

File: docs/source/delta-clustering.md [hunk 2]
```diff
@@ -87,6 +97,16 @@ OPTIMIZE table_name;
...
 
+In <Delta> 3.3 and above, you can force reclustering of all records in a table with the following syntax:
+
+```sql
+OPTIMIZE table_name FULL;
+```
+
+.. important:: Running `OPTIMIZE FULL` reclusters all existing data as necessary. For large tables that have not previously been clustered on the specified keys, this operation might take hours.
+
+Run `OPTIMIZE FULL` when you enable clustering for the first time or change clustering keys. If you have previously run `OPTIMIZE FULL` and there has been no change to clustering keys, `OPTIMIZE FULL` runs the same as `OPTIMIZE`. Always use `OPTIMIZE FULL` to ensure that data layout reflects the current clustering keys.
```
Comment (rahulsmahadev) [inline docs/source/delta-clustering.md:108]: `Run `OPTIMIZE FULL` when you enable clustering for the first time` is this needed?

Comment (zedtang) [reply docs/source/delta-clustering.md]: nope, removed.

**Review by rahulsmahadev — APPROVED** [review]

**Pull request merged**